src/main/java/de/doubleslash/keeptime/controller/Controller.java

```diff
@@ -51,44 +53,59 @@
    public Controller(final Model model, final DateProvider dateProvider) {
       this.model = model;
       this.dateProvider = dateProvider;
+
+      // initiate quicksaving
+      new Interval(QUICK_SAVE_INTERVAL).registerCallBack(() -> saveCurrentWork(0));
    }
 
    public void changeProject(final Project newProject) {
       changeProject(newProject, 0);
    }
 
    public void changeProject(final Project newProject, final long minusSeconds) {
-      final Work currentWork = model.activeWorkItem.get();
 
       final LocalDateTime now = dateProvider.dateTimeNow().minusSeconds(minusSeconds);
       final LocalDate dateNow = now.toLocalDate();
-      if (currentWork != null) {
-         currentWork.setEndTime(now);
-         if (currentWork.getNotes().isEmpty()) {
-            currentWork.setNotes("- No notes -");
-         }
 
-         final String time = DateFormatter
-               .secondsToHHMMSS(Duration.between(currentWork.getStartTime(), currentWork.getEndTime()).getSeconds());
-
-         LOG.info("You worked from '{}' to '{}' ({}) on project '{}' with notes '{}'", currentWork.getStartTime(),
-               currentWork.getEndTime(), time, currentWork.getProject().getName(), currentWork.getNotes());
-
-         // Save in db
-         model.getWorkRepository().save(currentWork);
-      }
+      saveCurrentWork(minusSeconds);
 
       // Start new work
       final Work work = new Work(dateNow, now, now.plusSeconds(minusSeconds), newProject, "");
 
       model.getPastWorkItems().add(work);
+
+      model.activeWorkItem.set(work);
+   }
+
+   public void saveCurrentWork(final long minusSeconds) {
+      final Work currentWork = model.activeWorkItem.get();
+
+      if (currentWork == null) {
+         return;
+      }
+      final LocalDateTime now = dateProvider.dateTimeNow().minusSeconds(minusSeconds);
+      final LocalDate dateNow = now.toLocalDate();
+
+      currentWork.setEndTime(now);
+      if (currentWork.getNotes().isEmpty()) {
+         currentWork.setNotes("- No notes -");
+      }
+
+      final String time = DateFormatter
+            .secondsToHHMMSS(Duration.between(currentWork.getStartTime(), currentWork.getEndTime()).getSeconds());
+
+      LOG.info("Saving Work from '{}' to '{}' ({}) on project '{}' with notes '{}'", currentWork.getStartTime(),
+            currentWork.getEndTime(), time, currentWork.getProject().getName(), currentWork.getNotes());
+
+      // Save in db
+      model.getWorkRepository().save(currentWork);
+
       if (currentWork != null && !dateNow.isEqual(currentWork.getCreationDate())) {
```

Comment:
this logic should not be in the save work logic.
also this will confuse the tool now, if a work was started on day A and now is day B, it will remove the active item from pastWorkItems, but it is still active. need to be rethinked how this is handeled. Guess this is solved if you move it back to changeProject